### PR TITLE
WPasswordEntry widget

### PR DESCRIPTION
**Pull request by timeopochin**
Adds WPasswordEntry, a text entry widget that hides the typed text replacing it with `****`

### CODE REVIEW

File: example_widgets.py [hunk 1]
```diff
@@ -27,6 +27,9 @@
     d.add(1, 5, "Auto complete:")
     d.add(15, 5, WAutoComplete(8, "fo", ["foo", "foobar", "bar", "car", "dar"]))
 
+    d.add(1, 6, "Password:")
```
Comment (pfalcon) [inline example_widgets.py:30]: This change should be in a separate commit.

File: picotui/widgets.py [hunk 2]
```diff
@@ -464,6 +465,15 @@ def show_line(self, l, i):
         self.attr_reset()
 
 
+class WPasswordEntry(WTextEntry):
+
+    def __init__(self, w, text):
```
Comment (pfalcon) [inline picotui/widgets.py]: If the constructor doesn't really do anything, why it's there at all?

File: picotui/widgets.py [hunk 3]
```diff
@@ -464,6 +465,15 @@ def show_line(self, l, i):
...
+        super().__init__(w, text)
+
+    def show_line(self, l, i):
+        super().show_line('*'*len(l), i)
```
Comment (pfalcon) [inline picotui/widgets.py]: Please use spaces around `*`. Please use double-quote (`"`) for strings.

**Review by pfalcon — CHANGES_REQUESTED** [review]
I try to keep Picotui lean and mean, and usually suggest to put additional widgets in seperate libraries. But if this is a 2-liner, why not?

But let's first make it 2-liner and follow the existing codebase conventions. Beyond the inline comments, please make sure that any fixup commit are squashed into the original commit, and please run `git log`, and follow the same commit message format for your commits.

Thanks!

**Comment by timeopochin** [comment]
No problem, I'm new to contributing, this is the first pull request I've ever done, I'll make those changes :) great project by the way, and looking forwards to 1.4/2.0

**Comment by timeopochin** [comment]
I've made the changes, hope this is right :)

**Comment by pfalcon** [comment]
Thanks for updates! I actually also meant update to commit message format to following existing format, but I did that myself. And while looking closer at it, I figured WPasswordEntry is to long, and shortened it to WPasswdEntry.

Merged with the above changes, thanks for the contribution!

**Comment by timeopochin** [comment]
Great, I'm glad I was able to contribute, I've found some bugs with the WTextEntry and WMultiEntry, whist testing it, I will open separate issues/fixing pull requests when I get the time :)

**Pull request closed without merging**